components/core/src/ffi/ir_stream/decoding_methods.tpp

```diff
@@ -0,0 +1,106 @@
+#ifndef FFI_IR_STREAM_DECODING_METHODS_TPP
+#define FFI_IR_STREAM_DECODING_METHODS_TPP
+
+// C++ standard libraries
+#include <string_view>
+#include <vector>
+
+// Project headers
+#include "../../ReaderInterface.hpp"
+#include "../encoding_methods.hpp"
+#include "decoding_methods.hpp"
+#include "protocol_constants.hpp"
+
+namespace ffi::ir_stream {
+    template <typename encoded_variable_t, typename ConstantHandler,
+              typename EncodedIntHandler, typename EncodedFloatHandler, typename DictVarHandler>
+    void generic_decode_message (const std::string& logtype,
+                                 const std::vector<encoded_variable_t>& encoded_vars,
+                                 const std::vector<std::string>& dict_vars,
+                                 ConstantHandler constant_handler,
+                                 EncodedIntHandler encoded_int_handler,
+                                 EncodedFloatHandler encoded_float_handler,
+                                 DictVarHandler dict_var_handler) {
+        size_t encoded_vars_length = encoded_vars.size();
+        size_t dict_vars_length = dict_vars.size();
+        size_t next_static_text_begin_pos = 0;
+
+        size_t dictionary_vars_ix = 0;
+        size_t encoded_vars_ix = 0;
+        for (size_t cur_pos = 0; cur_pos < logtype.length(); ++cur_pos) {
```

Comment:
Move logtype.length() out from the loop body and make it as a const variable (especially logtype.length() is multiple times in this function)